src/anemoi/transform/filters/q_to_r_height.py

```diff
@@ -0,0 +1,456 @@
+            AB = AB.upper()
+            if AB in PREDEFINED_AB.keys():
+                AB = PREDEFINED_AB[AB]
+            else:
+                KeyError(
+                    "%s is not in the list of predefined AB-coefficients. Possible options are %s."
+                    % (AB, ", ".join(PREDEFINED_AB.keys()))
+                )
+        if not isinstance(AB, dict):
+            TypeError("AB must be a string or a dictionary.")
+        self.A = np.array(AB["A"])
+        self.B = np.array(AB["B"])
+
+    def _get_pressure_at_heigh_level(
+        self,
+        temperature_at_model_levels: ekd.FieldList,
+        specific_humidity_at_model_levels: ekd.FieldList,
+        surface_pressure: ekd.Field
+    ):
+        return vertical.pressure_at_height_levels(
+            height = self.height,
+            t = temperature_at_model_levels,
+            q = specific_humidity_at_model_levels,
+            sp = surface_pressure,
+            A = self.A,
+            B = self.B,
+        )
+
+    def forward_transform(
+        self,
+        specific_humidity_at_height_level: ekd.Field,
+        temperature_at_height_level: ekd.Field,
+        surface_pressure: ekd.Field,
+        specific_humidity_at_model_levels: ekd.FieldList,
+        temperature_at_model_levels: ekd.FieldList,
+    ) -> Iterator[ekd.Field]:
+        """This will return the relative humidity along with temperature from specific humidity and temperature"""
+
+        # Make sure model levels are ordered ascending (highest level first):
+        specific_humidity_at_model_levels = specific_humidity_at_model_levels.order_by(level="ascending")
+        temperature_at_model_levels = temperature_at_model_levels.order_by(level="ascending")
+
+        pressure_at_height_level = self._get_pressure_at_heigh_level(
+            temperature_at_model_levels.to_numpy(),
+            specific_humidity_at_model_levels.to_numpy(),
+            surface_pressure.to_numpy()       
+        )
+        
+        # For now We need to go from qv --> td --> rh to take into account
+        # the mixed / ice phase when T ~ 0C / T < 0C
+        # See https://github.com/ecmwf/earthkit-meteo/issues/15
+        dewpoint_at_height_level = thermo.dewpoint_from_specific_humidity(
+            q = specific_humidity_at_height_level.to_numpy(), 
+            p = pressure_at_height_level
+        )
+        relative_humidity_at_height_level = thermo.relative_humidity_from_dewpoint(
+            t = temperature_at_height_level.to_numpy(),
+            td = dewpoint_at_height_level
+        )
+
+        # Return the fields
+        yield self.new_field_from_numpy(
+            relative_humidity_at_height_level,
+            template=specific_humidity_at_height_level,
+            param=self.relative_humidity_at_height_level,
+        )
+        yield temperature_at_height_level
+        #TODO Do we wan't to keep specific hum. when we have converted it?
+        yield specific_humidity_at_height_level  
```

Comment:
Just to flag @flyIchtus has a nice PR for controlling this better https://github.com/ecmwf/anemoi-transform/pull/100. We are planning to review during the next couple of weeks